storage/replica_data_iter_test.go

```diff
@@ -196,7 +194,18 @@ func TestReplicaDataIterator(t *testing.T) {
 	iter = newReplicaDataIterator(tc.rng.Desc(), tc.rng.store.Engine())
 	defer iter.Close()
 	if iter.Valid() {
-		t.Errorf("expected empty iteration; got first key %q", iter.Key())
+		// If the range is destroyed, only a tombstone key will be there.
+		k1, _, _, err := engine.MVCCDecodeKey(iter.Key())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		iter.Next()
+		if bytes.Compare(k1, keys.RaftTombstoneKey(tc.rng.Desc().RangeID)) != 0 || iter.Valid() {
+			t.Errorf("expected a destroyed replica to have only a tombstone key, but found more")
```

Comment:
sorry, i wasn't clear before; this error message should be broken into two separate errors:
```go
if tombstoneKey := keys.RaftTombstoneKey(tc.rng.Desc().RangeID); !bytes.Equal(k1, tombstoneKey) {
  t.Errorf("expected tombstone key %q but found %q", tombstoneKey, k1)
}

if iter.Valid() {
  t.Error("expected a destroyed replica to have only a tombstone, but found more")
}
```